src/guiguts/tools/levenshtein.py

```diff
@@ -0,0 +1,683 @@
+                    lines_output_count = -1
+                    break
+                error_start, error_end = make_into_strings(index_tuple, len(test_word))
+                start_rowcol = IndexRowCol(error_start)
+                end_rowcol = IndexRowCol(error_end)
+                file_line = maintext().get(
+                    error_start + " linestart", error_end + " lineend"
+                )
+                checker_dialog.add_entry(
+                    file_line,
+                    IndexRange(start_rowcol, end_rowcol),
+                    start_rowcol.col,
+                    end_rowcol.col,
+                )
+                lines_output_count += 1
+
+            if lines_output_count < 0:
+                checker_dialog.add_footer("  ...more")
+
+            # Add a blank line to separate this word-pair report from that of the
+            # next pair.
+            checker_dialog.add_footer("")
+
+        def make_into_strings(
+            index_tuple: tuple, hilite_length: int
+        ) -> tuple[str, str]:
+            """Utility function to generate error_start & error_end strings from int values
+
+            Args:
+                index_tuple: an integer line number and an integer column start index
+                hilite_length: integer length of string to be lighlighted on the line.
+
+            Return:
+                An error_start string (E.g. "18.3" and an error_end string (E.g. "18.9")
+            """
+
+            line_num = str(index_tuple[0])
+            col_indx = str(index_tuple[1])
+            error_start = f"{line_num}.{col_indx}"
+            col_indx = str(index_tuple[1] + hilite_length)
+            error_end = f"{line_num}.{col_indx}"
+
+            return error_start, error_end
+
+        def map_words_in_file(project_dict: ProjectDict) -> None:
+            """Spell check the currently loaded file and extract a list of suspect
+            words and a list of good words. Build a map of the line position index(es)
+            of each word and a map of the frequency of each word.
+
+            Outputs into a nonlocal list or dictionary:
+                List of suspect words - each an apparent spelling error.
+                Counts of those words (as a dictionary).
+                List of good words - each found in an English dictionary.
+                Counts of those words (as a dictionary).
+                Map of the line index(es) of each word and hilite start index on line.
+            """
+
+            # These must already exist (see enclosing function) before they are
+            # used here. They are the lists and dictionaries into which the results
+            # of this function are 'returned'.
+            nonlocal good_words
+            nonlocal suspect_words
+            nonlocal word_to_lines_map
+            nonlocal all_words_case_map, all_words_counts
+
+            # Will check spelling using SpellChecker methods from spell.py
+            spelling = SpellChecker()
```

Comment:
This can end up raising SpellChecker's `DictionaryNotFoundError` exception, so either needs to be caught here, or possibly the SpellChecker could be created higher up and passed as an argument.
If it was created during the `init` and stored as `self.spellchecker`, then the try/except at line 679 could remain, though it should use SpellChecker's exception, not the one created at the top of this file.

Reply:
Forgot to say - an easy way to trigger this is to change the language to `xyz` before running Levenshtein.